pkg/controller/secretshare/secretshare_controller.go

```diff
@@ -130,36 +136,59 @@ func (r *ReconcileSecretShare) copySecret(instance *ibmcpcsv1.SecretShare) bool
 	secretList := instance.Spec.Secretshares
 	requeue := false
 	for _, secretShare := range secretList {
+		running := true
 		secretName := secretShare.Secretname
 		secret, err := r.getSecret(secretName, ns)
 		if err != nil {
-			klog.Error(err)
-			requeue = true
-			continue
-		}
-		// Set SecretShare instance as the owner
-		if err := controllerutil.SetOwnerReference(instance, secret, r.scheme); err != nil {
-			klog.Error(err)
-			requeue = true
-			continue
-		}
-		if err := r.updateSecret(secret); err != nil {
-			klog.Error(err)
-			requeue = true
-			continue
-		}
-		for _, ns := range secretShare.Sharewith {
-			if err := r.ensureNs(ns.Namespace); err != nil {
+			if errors.IsNotFound(err) && instance.CheckStatus(ns+"/"+secretName, ibmcpcsv1.Running) {
+				if r.deleteCopiedSecret(secretName, secretShare) {
+					instance.UpdateStatus(ns+"/"+secretName, ibmcpcsv1.Failed)
+					requeue = true
+					continue
+				}
+				instance.RemoveStatus(ns + "/" + secretName)
```

Comment:
May I ask that why we do need to remove status when there is no err found in errors and the previous status is running?
Whatever we deleted copied secret successfully or not.

Reply:
The Controller will remove the status if the secret/configmap is not found and its status is running, which means the original secret/configmap has been deleted. The secretshare operator will delete the copied secret/configmap and clean up the status.